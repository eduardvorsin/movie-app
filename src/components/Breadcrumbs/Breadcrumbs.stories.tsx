import type { Meta, StoryObj } from '@storybook/react';
import Breadcrumbs from './Breadcrumbs';

const meta: Meta<typeof Breadcrumbs> = {
	title: 'UI/Breadcrumbs',
	tags: ['autodocs'],
	argTypes: {
		label: {
			description: 'Text description for the element',
		},
		className: {
			description: 'Additional classes for the component'
		},
		isNavigation: {
			description: 'Allows you to select whether the element will be a navigation block or a regular block',
		},
		testId: {
			description: 'Id for testing the component',
		},
		maxItems: {
			description: 'Defines the maximum number of visible elements',
		},
		separator: {
			description: 'The character to be used as a separator',
		},
		truncationWidth: {
			description: 'Sets the maximum width of the navigation elements at which the text will be cut off',
		},
		target: {
			control: false,
			description: 'How to display a page when clicking on a link',
		},
		lastItemLabel: {
			description: 'The label for the last element in the navigation',
		},
	},
	component: Breadcrumbs,
	parameters: {
		docs: {
			description: {
				component: 'The Breadcrumbs component is a navigation element of the website or application interface that provides users with information about the current location on the site relative to its structure.',
			},
		},
		nextjs: {
			navigation: {
				pathname: '/en/movies/9999',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Breadcrumbs>;

export const Default: Story = {
	args: {
		label: 'navigation',
		maxItems: 2,
	},
	parameters: {
		docs: {
			description: {
				story: 'The default state of the component',
			},
		},
	},
};

export const TrunctionWidth: Story = {
	args: {
		label: 'navigation',
		maxItems: 2,
		truncationWidth: 50,
	},
	parameters: {
		docs: {
			description: {
				story: 'The value in pixels after which the text in the navigation elements will be cropped',
			},
		},
	},
};

export const LastItemLabel: Story = {
	args: {
		label: 'navigation',
		maxItems: 2,
		lastItemLabel: 'Star Wars',
	},
	parameters: {
		docs: {
			description: {
				story: 'A text label that will be displayed instead of the last navigation element',
			},
		},
	},
};

export const Separator: Story = {
	args: {
		label: 'navigation',
		maxItems: 2,
		separator: '>',
	},
	parameters: {
		docs: {
			description: {
				story: 'Custom symbol for the element separator',
			},
		},
	},
};