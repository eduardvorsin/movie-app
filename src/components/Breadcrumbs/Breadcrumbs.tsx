'use client';
import { useTranslation } from '@/i18n/client';
import { Locales, fallbackLng, locales } from '@/i18n/settings';
import { useParams, usePathname } from 'next/navigation';
import { AnchorHTMLAttributes, MouseEventHandler, useState } from 'react';
import BreadcrumbsItem from './BreadcrumbsItem/BreadcrumbsItem';
import { navigationRoutes } from 'src/constants';
import { toCapitalizeCase } from '@/helpers/toCapitalizeCase/toCapitalizeCase';
import { GeneralProps } from '@/types/shared';

type Props = {
	label: string,
	isNavigation?: boolean,
	maxItems?: number,
	separator?: string,
	truncationWidth?: number,
	target?: AnchorHTMLAttributes<HTMLButtonElement>['target'],
	lastItemLabel?: string,
} & GeneralProps;

export default function Breadcrumbs({
	label,
	className,
	isNavigation = true,
	maxItems = +Infinity,
	testId,
	separator = '/',
	truncationWidth,
	target,
	lastItemLabel,
	...props
}: Props) {
	const lang = useParams()?.lang as Locales ?? fallbackLng;
	const { t } = useTranslation(lang);
	const [isExpanded, setIsExpanded] = useState<boolean>(false);

	const parts = usePathname().split('/').slice(1);
	const segments = parts
		.map((segment, index) => {
			let localizedSegment = segment;
			if (locales.includes(segment as Locales)) {
				localizedSegment = t('breadcrumbs.home');
			}
			if ((Object.keys(navigationRoutes)).includes(segment)) {
				localizedSegment = t(`breadcrumbs.${segment}`);
			}

			return {
				label: toCapitalizeCase(localizedSegment),
				value: `/${parts.slice(0, index + 1).join('/')}`,
			};
		});

	if (lastItemLabel) {
		segments[segments.length - 1].label = lastItemLabel;
	}

	const clickHandler: MouseEventHandler<HTMLButtonElement> = () => {
		setIsExpanded(true);
	}

	const Wrapper = isNavigation ? 'nav' : 'div';
	const totalItems = segments.map((segment, index) => (
		<BreadcrumbsItem
			key={segment.label}
			className={index === segments.length - 1 ? 'pointer-events-none' : ''}
			href={index === segments.length - 1 ? '' : segment.value}
			truncationWidth={truncationWidth}
			target={target}
			aria-current={index === segments.length - 1 ? 'page' : undefined}
		>
			{segment.label}
		</BreadcrumbsItem >
	));

	let visibleItems;
	if (maxItems >= totalItems.length || totalItems.length < 3 || isExpanded) {
		visibleItems = totalItems;
	} else {
		const elipsisButton: JSX.Element = (
			<button
				className='no-underline text-blue-500 hover:text-blue-600 hover:underline active:text-blue-700 transition-colors duration-150'
				onClick={clickHandler}
			>
				<span>...</span>
				<span className='sr-only'>{t('elipsisLabel')}</span>
			</button>
		);

		visibleItems = [
			totalItems[0],
			elipsisButton,
			totalItems[totalItems.length - 1],
		];
	}

	return (
		<Wrapper
			className={className}
			data-testid={testId}
			aria-label={label}
			{...props}
		>
			<ol className='flex items-center flex-wrap gap-y-1'>
				{visibleItems.map((child, index) => (
					<li
						key={index}
						className='text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150'
						data-separator={separator}
					>
						{child}
					</li>
				))}
			</ol>
		</Wrapper>
	);
};