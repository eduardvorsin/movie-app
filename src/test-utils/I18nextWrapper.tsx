import React, { ReactNode } from 'react';
import i18next from '@/i18n/client';
import { I18nextProvider } from 'react-i18next';

type Props = {
	children: ReactNode,
}

const I18nextWrapper = ({
	children,
}: Props) => {
	return (
		<I18nextProvider i18n={i18next}>
			{children}
		</I18nextProvider>
	);
};

export default I18nextWrapper;