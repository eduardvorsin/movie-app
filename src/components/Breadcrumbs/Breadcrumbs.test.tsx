import { render, screen } from "@testing-library/react";
import Breadcrumbs from "./Breadcrumbs";
import i18next from "@/i18n/client";
import I18nextWrapper from "@/test-utils/I18nextWrapper";

const mockUseParams = jest.fn();
const mockUsePathname = jest.fn();
jest.mock('next/navigation', () => ({
	useParams: () => mockUseParams(),
	usePathname: () => mockUsePathname(),
}));

describe('Breadcrumbs tests', () => {
	it('is rendered correctly', async () => {
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/movies/9999');

		render(
			<Breadcrumbs
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-breadcrumbs')).toBeInTheDocument();
	});

	it('if the number of elements is less than 3, then all the elements should be rendered', async () => {
		mockUseParams.mockReturnValue({ lang: 'en' });
		mockUsePathname.mockReturnValue('/en/movies');

		render(
			<Breadcrumbs
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>
		);

		expect(await screen.findAllByRole<HTMLAnchorElement>('link')).toHaveLength(2);
	});

	it('if maxItems is more than the number of elements, then it should render all the elements', async () => {
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/movies/9999');

		render(
			<Breadcrumbs
				maxItems={5}
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>
		);

		expect(await screen.findAllByRole<HTMLAnchorElement>('link')).toHaveLength(3);
	});

	it('if maxItems is less than the number of elements, then the first and last element should be rendered', async () => {
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/tv/9999/seasons');

		render(
			<Breadcrumbs
				maxItems={2}
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>
		);

		expect(await screen.findByText<HTMLSpanElement>(/home page/i)).toBeInTheDocument();
		expect(screen.getByText<HTMLSpanElement>(/seasons/i)).toBeInTheDocument();
	});

	it('is a basic snapshot', async () => {
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/movies/9999');

		render(
			<Breadcrumbs
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-breadcrumbs')).toMatchSnapshot();
	});

	it('a snapshot with a truncationWidth of 100', async () => {
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/movies/9999');

		render(
			<Breadcrumbs
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
				truncationWidth={100}
			/>
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-breadcrumbs')).toMatchSnapshot();
	});

	it('a snapshot when all the elements are rendered', async () => {
		mockUseParams.mockReturnValue({ lang: 'en' });
		mockUsePathname.mockReturnValue('/en/movies');

		render(
			<Breadcrumbs
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-breadcrumbs')).toMatchSnapshot();
	});

	it('snapshot when the first and last element were rendered', async () => {
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/tv/9999/seasons');

		render(
			<Breadcrumbs
				maxItems={2}
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-breadcrumbs')).toMatchSnapshot();
	});

	it('a snapshot with isNavigation set to true', async () => {
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/movies/9999');

		render(
			<Breadcrumbs
				isNavigation
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>
		);

		expect(await screen.findByTestId<HTMLDivElement>('test-breadcrumbs')).toMatchSnapshot();
	});

	it('snapshot with lastItemLabel', async () => {
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/movies/9999');

		render(
			<Breadcrumbs
				isNavigation
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
				lastItemLabel='Star Wars'
			/>
		);
		expect(await screen.findByTestId<HTMLDivElement>('test-breadcrumbs')).toMatchSnapshot();
	});
});

describe('Breadcrumbs integration tests', () => {
	it('localization into English works correctly', async () => {
		await i18next.changeLanguage('en');
		const translation = new RegExp(i18next.t('breadcrumbs.elipsisLabel'));
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/movies/9999');

		render(
			<Breadcrumbs
				maxItems={2}
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLSpanElement>(translation)).toBeInTheDocument();
	});

	it('localization into Russian works correctly', async () => {
		await i18next.changeLanguage('ru');
		const translation = new RegExp(i18next.t('breadcrumbs.elipsisLabel'));
		mockUseParams.mockReturnValue({ lang: 'ru', id: '9999' });
		mockUsePathname.mockReturnValue('/ru/movies/9999');

		render(
			<Breadcrumbs
				maxItems={2}
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLSpanElement>(translation)).toBeInTheDocument();
	});

	it('is a snapshot with English localization', async () => {
		await i18next.changeLanguage('en');
		mockUseParams.mockReturnValue({ lang: 'en', id: '9999' });
		mockUsePathname.mockReturnValue('/en/movies/9999');

		render(
			<Breadcrumbs
				maxItems={2}
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByTestId<HTMLDivElement>('test-breadcrumbs')).toMatchSnapshot();
	});

	it('is a snapshot with Russian localization', async () => {
		await i18next.changeLanguage('ru');
		mockUseParams.mockReturnValue({ lang: 'ru', id: '9999' });
		mockUsePathname.mockReturnValue('/ru/movies/9999');

		render(
			<Breadcrumbs
				maxItems={2}
				label='test-breadcrumbs'
				testId='test-breadcrumbs'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByTestId<HTMLDivElement>('test-breadcrumbs')).toMatchSnapshot();
	});
});