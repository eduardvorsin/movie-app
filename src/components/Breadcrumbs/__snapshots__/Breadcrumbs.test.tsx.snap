// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs integration tests is a snapshot with English localization 1`] = `
<nav
  aria-label="test-breadcrumbs"
  data-testid="test-breadcrumbs"
>
  <ol
    class="flex items-center flex-wrap gap-y-1"
  >
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en"
      >
        <span
          class=""
        >
          Home Page
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <button
        class="no-underline text-blue-500 hover:text-blue-600 hover:underline active:text-blue-700 transition-colors duration-150"
      >
        <span>
          ...
        </span>
        <span
          class="sr-only"
        >
          expand all links
        </span>
      </button>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-current="page"
        aria-disabled="true"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 pointer-events-none"
        href=""
        role="link"
      >
        <span
          class=""
        >
          9999
        </span>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs integration tests is a snapshot with Russian localization 1`] = `
<nav
  aria-label="test-breadcrumbs"
  data-testid="test-breadcrumbs"
>
  <ol
    class="flex items-center flex-wrap gap-y-1"
  >
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/ru"
      >
        <span
          class=""
        >
          Главная Страница
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <button
        class="no-underline text-blue-500 hover:text-blue-600 hover:underline active:text-blue-700 transition-colors duration-150"
      >
        <span>
          ...
        </span>
        <span
          class="sr-only"
        >
          развернуть все ссылки
        </span>
      </button>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-current="page"
        aria-disabled="true"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 pointer-events-none"
        href=""
        role="link"
      >
        <span
          class=""
        >
          9999
        </span>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs tests a snapshot when all the elements are rendered 1`] = `
<nav
  aria-label="test-breadcrumbs"
  data-testid="test-breadcrumbs"
>
  <ol
    class="flex items-center flex-wrap gap-y-1"
  >
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en"
      >
        <span
          class=""
        >
          Home Page
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-current="page"
        aria-disabled="true"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 pointer-events-none"
        href=""
        role="link"
      >
        <span
          class=""
        >
          Movies
        </span>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs tests a snapshot with a truncationWidth of 100 1`] = `
<nav
  aria-label="test-breadcrumbs"
  data-testid="test-breadcrumbs"
>
  <ol
    class="flex items-center flex-wrap gap-y-1"
  >
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en"
        style="max-width: 100px;"
      >
        <span
          class="truncate"
        >
          Home Page
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en/movies"
        style="max-width: 100px;"
      >
        <span
          class="truncate"
        >
          Movies
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-current="page"
        aria-disabled="true"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 pointer-events-none"
        href=""
        role="link"
        style="max-width: 100px;"
      >
        <span
          class="truncate"
        >
          9999
        </span>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs tests a snapshot with isNavigation set to true 1`] = `
<nav
  aria-label="test-breadcrumbs"
  data-testid="test-breadcrumbs"
>
  <ol
    class="flex items-center flex-wrap gap-y-1"
  >
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en"
      >
        <span
          class=""
        >
          Home Page
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en/movies"
      >
        <span
          class=""
        >
          Movies
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-current="page"
        aria-disabled="true"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 pointer-events-none"
        href=""
        role="link"
      >
        <span
          class=""
        >
          9999
        </span>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs tests is a basic snapshot 1`] = `
<nav
  aria-label="test-breadcrumbs"
  data-testid="test-breadcrumbs"
>
  <ol
    class="flex items-center flex-wrap gap-y-1"
  >
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en"
      >
        <span
          class=""
        >
          Home Page
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en/movies"
      >
        <span
          class=""
        >
          Movies
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-current="page"
        aria-disabled="true"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 pointer-events-none"
        href=""
        role="link"
      >
        <span
          class=""
        >
          9999
        </span>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs tests snapshot when the first and last element were rendered 1`] = `
<nav
  aria-label="test-breadcrumbs"
  data-testid="test-breadcrumbs"
>
  <ol
    class="flex items-center flex-wrap gap-y-1"
  >
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en"
      >
        <span
          class=""
        >
          Home Page
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <button
        class="no-underline text-blue-500 hover:text-blue-600 hover:underline active:text-blue-700 transition-colors duration-150"
      >
        <span>
          ...
        </span>
        <span
          class="sr-only"
        >
          expand all links
        </span>
      </button>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-current="page"
        aria-disabled="true"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 pointer-events-none"
        href=""
        role="link"
      >
        <span
          class=""
        >
          Seasons
        </span>
      </a>
    </li>
  </ol>
</nav>
`;

exports[`Breadcrumbs tests snapshot with lastItemLabel 1`] = `
<nav
  aria-label="test-breadcrumbs"
  data-testid="test-breadcrumbs"
>
  <ol
    class="flex items-center flex-wrap gap-y-1"
  >
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en"
      >
        <span
          class=""
        >
          Home Page
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-disabled="false"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 "
        href="/en/movies"
      >
        <span
          class=""
        >
          Movies
        </span>
      </a>
    </li>
    <li
      class="text-100 after:content-[attr(data-separator)] last:after:content-none after:mx-2 after:text-center after:text-100 dark:after:text-dark-neutral-1000 after:text-neutral-1000 transition-colors duration-150 after:transition-colors after:duration-150"
      data-separator="/"
    >
      <a
        aria-current="page"
        aria-disabled="true"
        class="no-underline  hover:underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200  inline-flex text-200 pointer-events-none"
        href=""
        role="link"
      >
        <span
          class=""
        >
          Star Wars
        </span>
      </a>
    </li>
  </ol>
</nav>
`;
